src/Microsoft.Sbom.Api/Executors/Spdx3SerializationStrategy.cs

```diff
@@ -0,0 +1,135 @@
+public class Spdx3SerializationStrategy : IJsonSerializationStrategy
+{
+    public void AddToFilesSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config)
+    {
+        elementsSupportingConfigs.Add(config);
+    }
+
+    public void AddToPackagesSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config)
+    {
+        elementsSupportingConfigs.Add(config);
+    }
+
+    public bool AddToRelationshipsSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config)
+    {
+        return true;
+    }
+
+    public void AddToExternalDocRefsSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config)
+    {
+        elementsSupportingConfigs.Add(config);
+    }
+
+    public async Task<List<FileValidationResult>> WriteJsonObjectsToSbomAsync(
+        ISbomConfig sbomConfig,
+        string spdxManifestVersion,
+        IJsonArrayGenerator<FileArrayGenerator> fileArrayGenerator,
+        IJsonArrayGenerator<PackageArrayGenerator> packageArrayGenerator,
+        IJsonArrayGenerator<RelationshipsArrayGenerator> relationshipsArrayGenerator,
+        IJsonArrayGenerator<ExternalDocumentReferenceGenerator> externalDocumentReferenceGenerator)
+    {
+        fileArrayGenerator.SbomConfig = sbomConfig;
+        packageArrayGenerator.SbomConfig = sbomConfig;
+        relationshipsArrayGenerator.SbomConfig = sbomConfig;
+        externalDocumentReferenceGenerator.SbomConfig = sbomConfig;
+
+        fileArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        packageArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        relationshipsArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        externalDocumentReferenceGenerator.SpdxManifestVersion = spdxManifestVersion;
+
+        // Holds the SPDX IDs of all the elements that have been written to the SBOM. Used for deduplication.
+        var elementsSpdxIdList = new HashSet<string>();
+
+        WriteContext(sbomConfig);
+
+        sbomConfig.JsonSerializer.StartJsonArray(Constants.SPDXGraphHeaderName);
+
+        // Files section
+        var generateResult = await fileArrayGenerator.GenerateAsync();
+        WriteElementsToSbom(generateResult, elementsSpdxIdList);
+
+        // Packages section
+        var packagesGenerateResult = await packageArrayGenerator.GenerateAsync();
+        generateResult.Errors.AddRange(packagesGenerateResult.Errors);
+        WriteElementsToSbom(packagesGenerateResult, elementsSpdxIdList);
+
+        // External Document Reference section
+        var externalDocumentReferenceGenerateResult = await externalDocumentReferenceGenerator.GenerateAsync();
+        generateResult.Errors.AddRange(externalDocumentReferenceGenerateResult.Errors);
+        WriteElementsToSbom(externalDocumentReferenceGenerateResult, elementsSpdxIdList);
+
+        // Relationships section
+        var relationshipGenerateResult = await relationshipsArrayGenerator.GenerateAsync();
```

Comment:
Variable name didn't change when `GenerateResult` type was renamed to `GenerationResult`